### PR TITLE
Coverage needs develop, actions update

### PR DESCRIPTION
**Pull request by thatch**

### CODE REVIEW

File: codemod_tox/core.bak
Comment (amjith) [inline codemod_tox/core.bak:1]: Why is it not .py? Mistake?

File: codemod_tox/core.bak [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        assert s[0] == "{"
+        assert s[-1] == "}"
+        return ToxOptions(tuple(i.strip() for i in s[1:-1].split(",")))
+
+    def __str__(self) -> str:
+        return "{%s}" % ",".join(self.options)
+
+
+@dataclass
+class ToxEnv:
```
Comment (amjith) [inline codemod_tox/core.bak]: Inherit ToxBase?

**Comment by thatch** [comment]
The .bak was an experiment to preserve inconsistent whitespace, but it
ended up being a lot of complexity. I didn't mean to include it here (but
will probably save it as a dormant branch just so I don't lose it)

File: codemod_tox/core.bak [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    for i, x in enumerate(seq):
+        yield x, i==last_idx
+
+def _common_prefix(a: str, b: str) -> str:
+    buf = ""
+    for c1, c2 in zip(a, b):
+        if c1 != c2:
+            break
+        buf += c1
+    return buf
```
Comment (amjith) [inline codemod_tox/core.bak]: Aside: Went down the rabbit hole of why we can't use sets for this and then why are sets not order preserving.

File: codemod_tox/core.bak [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+    def __str__(self) -> str:
+        buf = ""
+        default_suffix = "\n"
+        
+        for item, last in _marklast(self.envs):
+            buf += str(item)
+            if not last:
+                if "," in item.suffix:
+                    default_suffix = ", "
```
Comment (amjith) [inline codemod_tox/core.bak]: Are we planning to use a comma if the user had a comma separated list and only fallback to newlines if there were no commas to being with?

**Comment by thatch** [comment]
Indeed, newline unless there's a comma is what the experiment does. I'm not
super happy with that (say, what if you use comma with no space
consistently, or only have one item now).

I'm only intending to output newlines for MVP,.and you can use
list(a.all()) == list(b.all()) to check for whether it's changed before
modifying the initial entry

**Review by amjith — APPROVED** [review]

**Pull request merged**